cineast-api/src/main/java/org/vitrivr/cineast/api/APIEndpoint.java

```diff
@@ -97,103 +198,210 @@ private static Service dispatchService(boolean secure) {
+            System.exit(100);
+        }
+
         return service;
     }
 
+    private int validatePort(boolean secure, APIConfig config) {
+        int port = secure ? config.getHttpsPort() : config.getHttpPort();
+        if(port <= 0 || port >= 65535) {
+            LOGGER.warn("The specified port {} is not valid. Fallback to default port.", port);
+            return -1;
+        }
+        return port;
+    }
+
     /**
      * Starts the RESTful / WebSocket API.
      */
-    public static void start() {
+    public void start() {
         /* Start insecure HTTP connection. */
         if (Config.sharedConfig().getApi().getEnableRest() || Config.sharedConfig().getApi().getEnableWebsocket()) {
             http = dispatchService(false);
-            http.init();
-            http.awaitInitialization();
         }
 
         if (Config.sharedConfig().getApi().getEnableRestSecure() || Config.sharedConfig().getApi().getEnableWebsocketSecure()) {
             https = dispatchService(true);
-            https.init();
-            https.awaitInitialization();
         }
     }
 
     /**
      * Stops the RESTful / WebSocket API.
      */
     public static void stop() {
-        if (Config.sharedConfig().getApi().getEnableRest() || Config.sharedConfig().getApi().getEnableWebsocket()) {
+        if(instance != null){
+            instance.shutdown();
+        }
+    }
+
+    public void shutdown() {
+        if (Config.sharedConfig().getApi().getEnableRest() || Config.sharedConfig().getApi()
+                .getEnableWebsocket()) {
             http.stop();
         }
-        if (Config.sharedConfig().getApi().getEnableRestSecure() || Config.sharedConfig().getApi().getEnableWebsocketSecure()) {
+        if (Config.sharedConfig().getApi().getEnableRestSecure() || Config.sharedConfig().getApi()
+                .getEnableWebsocketSecure()) {
             https.stop();
         }
     }
 
+    /**
+     * Registers all routes of a single {@link DocumentedRestOperation} to the {@link Javalin} service.
+     * 
+     * @param config
+     * @param route
+     * @param service
+     */
+    private void registerRoute(APIConfig config, final DocumentedRestOperation<?, ?> route, final Javalin service){
+        route.supportedMethods().forEach(m -> {
+            RestHttpMethod method = (RestHttpMethod)m;
+
+            Handler handler = route;
+
+            if(config.getEnableLiveDoc()) {
+                //Document handler
+                OpenApiDocumentation document = OpenApiBuilder.document();
+
+                //Default route documentation
+                document.operation(operation -> {
+                    operation.description(route.getDescription(method));
+                    operation.summary(route.getSummary(method));
+                    operation.operationId(String.format("%s#%s", route.getClass().getName(), method.toString()));
+                    operation.addTagsItem(namespace());
```

Comment:
I'd like to see a todo here, as there is certainly a more sophisticated OAS tag system than this one.